### DISCUSSION TITLE
How to access server context set by middleware

### DISCUSSION THREAD

**Discussion by aheissenberger** [wakujs/waku#461] [Q&A] [answered]
1. How can I add a middleware - e.g. for reading Authentication Layer and add Content to a server context
2. How can I access the server context in a React Server Component

I found the example `08_cookies` but I use `createPages` in `entries.tsx` and I found no way to add a Middleware without creating a `start.js` or `dev.js` and modify the builder for e.g. `serve-aws-lambda.ts`. 

Would it be possible to add this to `waku.config.ts`?

**Comment by dai-shi** [comment]
This is also what I want to cover with #329.
`08_cookies` will be updated then without needing `(dev|start).js`.

There's one challenge though. As we want to avoid node dependency, we don't use `AsyncLocalStorage`. Currently, we use `this` for server context, but that's not very useful (only accessible from `entries.tsx` and server actions. And, `this` is very confusing.) I will look into `React.cache` for the solution, hopefully possibly.

**Reply by aheissenberger** [reply]
I did it with `React.cache` in my #462 pull request

**Reply by dai-shi** [reply]
Nice. 

@himself65 also wanted to avoid `this` hack.

**Reply by dai-shi** [reply]
We used to have `getContext`, which is removed in #213.
I will re-implement it with `React.cache`.
As it's going to be a breaking change, it will be in v0.20.
Please be patient.

**Reply by aheissenberger** [reply, marked as answer]
With #513 middleware and server context is supported. An implementation is the example [examples/08_cookie](examples/08_cookies)
